typescript/infra/scripts/secret-rpc-urls/set-rpc-urls.ts

```diff
@@ -0,0 +1,105 @@
+import { confirm } from '@inquirer/prompts';
+import { ethers } from 'ethers';
+
+import {
+  getSecretRpcEndpoints,
+  getSecretRpcEndpointsLatestVersionName,
+  secretRcpEndpointsExist,
+  setSecretRpcEndpoints,
+} from '../../src/agents/index.js';
+import { disableGCPSecretVersion } from '../../src/utils/gcloud.js';
+import { getArgs, withChain, withRpcUrls } from '../agent-utils.js';
+
+async function testProviders(rpcUrlsArray: string[]): Promise<boolean> {
```

Comment:
Just realizing we won't want to call this function for any non-EVM chains - you can use https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/mo/rpc-secrets-scripts/typescript/infra/src/utils/utils.ts#L265 to figure out whether a chain is Ethereum based or not

Reply:
Good spot